lib/test-file.js

```diff
@@ -0,0 +1,162 @@
+// Based on
+// https://github.com/smikes/test262-parser/blob/04902c58d2e2d1b452b34754924b8e75f2e5f3ac/lib/parser.js
+// Copyright (C) 2014, Microsoft Corporation. All rights reserved.
+// This code is governed by the BSD License found in the LICENSE-MICROSOFT.txt
+'use strict';
+
+const yaml = require('js-yaml');
+const yamlStart = '/*---';
+const yamlEnd = '---*/';
+
+/**
+ * Extract copyright message
+ *
+ * @method extractCopyright
+ * @param {Test262File} file - file object
+ * @return {string} the copyright string extracted from contents
+ * @private
+ */
+function extractCopyright(test262File) {
+  let lines = test262File.contents.split(/\r|\n/);
+
+  for (let line of lines) {
+    // The very first line may be a hashbang, so look at
+    // all lines until reaching a line that looks close
+    // enough to copyright
+    let result = /^(?:(?:\/\/.*)*)/.exec(line);
```

Comment:
I missed this RegExp during my first review, but it seems kind of wonky to me. We should revisit it later.